packages/rocketchat-livechat/server/hooks/saveAnalyticsData.js

```diff
@@ -0,0 +1,45 @@
+RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
+	// skips this callback if the message was edited
+	if (!message || message.editedAt) {
+		return message;
+	}
+
+	// check if room is livechat
+	if (room.t !== 'l') {
+		return message;
+	}
+
+	Meteor.defer(() => {
+		const now = new Date();
+		let analyticsData;
+
+		// if the message has a token, it was sent by the visitor
+		if (!message.token) {
```

Comment:
Let's improve this code...  you can add something like this above the `Meteor.defer()`:

```
if (message.token) {
    return message;
}
```

This change will improve the readability of the rest of the code as well.

Reply:
I guess it has to be this way, because even if `message.token` is `true`, we still call the method `saveAnalyticsDataByRoomId` to update message timestamps (since we have merged other method into one).